src/instantsend/instantsend.cpp

```diff
@@ -419,15 +419,17 @@ void CInstantSendManager::ProcessInstantSendLock(NodeId from, PeerManager& peerm
         mempool.AddTransactionsUpdated(1);
     }
 
+    MessageProcessingResult ret{/*is_masternode=*/m_signer != nullptr};
```

Comment:
PeerManager knows already about `is_masternode` or not; see implementation:
```
    const bool is_masternode = m_mn_activeman != nullptr;
```

Consider refactoring `RequestObject`, `AskPeersForTransaction` instead to drop this flag completely and simplify logic. Consider to move [this commit](https://github.com/dashpay/dash/pull/6815/commits/bc9e9194d5bd59a5e3543a48180ad93be5670457) and next a7f0a769ce7df611de09bf8b3fdcc9b869d9d55b and related refactorings to new PR, they seems a bit out-of-scope

Reply:
Addressed in https://github.com/dashpay/dash/pull/6821, dropped from PR.